python/pyarrow/_dataset.pyx

```diff
@@ -426,6 +519,45 @@ cdef class FileFragment(Fragment):
         return FileFormat.wrap(self.file_fragment.format())
 
 
+cdef class ParquetFileFragment(FileFragment):
+    """A Fragment representing a parquet file."""
+
+    cdef:
+        CParquetFileFragment* parquet_file_fragment
+
+    cdef void init(self, const shared_ptr[CFragment]& sp):
+        FileFragment.init(self, sp)
+        self.parquet_file_fragment = <CParquetFileFragment*> sp.get()
+
+    @property
+    def row_groups(self):
+        row_groups = set(self.parquet_file_fragment.row_groups())
+        if len(row_groups) != 0:
+            return row_groups
+        return None
+
+    def get_row_group_fragments(self):
```

Comment:
Conceptually, I would find it more logical to pass a `filter` here (similarly as how a filter is passed in the `Dataset.get_fragments()`)

Reply:
The `filter` is already specified in `Dataset.get_fragments()`. I'll add an optional filter here which will be `and`ed with the original filter.